decidim-proposals/spec/lib/decidim/content_parsers/proposal_parser_spec.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  module ContentParsers
+    describe ProposalParser do
+      let(:organization) { create(:organization) }
+      let(:feature) { create(:proposal_feature, organization: organization) }
+      let(:context) { { current_organization: organization } }
+      let!(:parser) { Decidim::ContentParsers::ProposalParser.new(content, context) }
+
+      describe "on parse" do
+        subject { parser.rewrite }
+
+        context "when content is nil" do
+          let(:content) { nil }
+
+          it { is_expected.to eq("") }
+          it "has empty metadata" do
+            subject
+            expect(parser.metadata).to be_a(Decidim::ContentParsers::ProposalParser::Metadata)
+            expect(parser.metadata.linked_proposals).to eq([])
+          end
+        end
+
+        context "when content is empty string" do
+          let(:content) { "" }
+
+          it { is_expected.to eq("") }
+          it "has empty metadata" do
+            subject
+            expect(parser.metadata).to be_a(Decidim::ContentParsers::ProposalParser::Metadata)
```

Comment:
Maybe the subject should be `parser.metadata`?

Reply:
But there are many assertions that use the current subject, like the one at line 30 in the code snippet you are referencing...